src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java

```diff
@@ -515,8 +524,10 @@ public Void caseTemplatizedType(TemplatizedType type) {
           }
           emit(templateTypeName);
           emit("<");
-          for (JSType tmplType : type.getTemplateTypes()) {
-            visitType(tmplType);
+          UnmodifiableIterator<JSType> it = type.getTemplateTypes().iterator();
+          while (it.hasNext()) {
+            visitType(it.next());
+            if (it.hasNext()) emit(",");
```

Comment:
these are two different things - one is a declaration of a parameterized type (T) and the other is providing a type parameter (T => string).
I tried merging these together in one function, but it is pretty awkward since for the first case you need `getTemplateTypesMap().getTemplateKeys()` which is an `Iterable<TemplateType>` and the other you need `getTemplateTypes()` which is an `Iterable<JSType>` which just yields a string, not something we can visit.
I think it's just awkward closure API showing up.